modules/infra/datapath/gr_control_output.h

```diff
@@ -23,6 +24,10 @@ typedef void (*control_output_cb_t)(struct rte_mbuf *);
 GR_MBUF_PRIV_DATA_TYPE(control_output_mbuf_data, {
 	control_output_cb_t callback;
 	clock_t timestamp;
+	// Node specific data
+	union {
+		struct rte_icmp_hdr;
+	};
```

Comment:
That is not a good idea to have icmp references in a generic control output node data. I would prefer a generic buffer that will be mapped to something else via the callback.